ui/src/main/java/com/adyen/checkout/ui/internal/Buttons.kt

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright (c) 2025 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 16/4/2025.
+ */
+
+package com.adyen.checkout.ui.internal
+
+import androidx.annotation.RestrictTo
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Button
+import androidx.compose.material3.ButtonDefaults.buttonColors
+import androidx.compose.material3.CircularProgressIndicator
+import androidx.compose.material3.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.unit.dp
+import com.adyen.checkout.ui.theme.AdyenCheckoutTheme as Theme
+
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+@Composable
+fun PrimaryButton(
+    onClick: () -> Unit,
+    text: String,
+    modifier: Modifier = Modifier,
+    icon: Int? = null,
+    isLoading: Boolean = false,
+) {
+    val buttonStyles = AdyenCheckoutTheme.elements.buttons
+    AdyenButton(
+        onClick = onClick,
+        text = text,
+        modifier = modifier,
+        icon = icon,
+        isLoading = isLoading,
+        cornerRadius = buttonStyles.cornerRadius ?: AdyenCheckoutTheme.elements.cornerRadius,
+        style = ButtonDefaults.primaryButtonStyle(buttonStyles.primary),
+    )
+}
+
+@Preview(showBackground = true)
```

Comment:
It would be nice to have a preview where we pass an icon as well, as an example of how it looks. Doesn't have to be this one.

Reply:
After some discussion we decided to remove icons as we don't have them in the design. We could always add them later if really needed